### PR TITLE
As/saving api

### PR DESCRIPTION
**Pull request by asinghvi17**

### CODE REVIEW

**Comment by asinghvi17** [comment]
Not sure why the CI is failing...it works fine on my machine.

**Comment by asinghvi17** [comment]
I guess it works now?  Not sure what was going on there.

TODOS:

- Create pre-eval and post-eval hooks

File: src/io.jl [hunk 1]
```diff
@@ -263,7 +281,7 @@ function record_examples(
         1
     end
     @info("starting from index $start")
-    AbstractPlotting.set_theme!(resolution = resolution)
+    setresolution!(resolution)
```
Comment (SimonDanisch) [inline src/io.jl]: This is a bit confusing... I find set theme clearer here, since we also reset the whole theme back to it's defaults...

Comment (asinghvi17) [reply src/io.jl]: Makes sense - I've changed the function name.

**Comment by asinghvi17** [comment]
To this PR, I've added an experimental Github action which will deploy documentation not only on pushes to master, but also on PRs.  We'll see if it works!

**Comment by asinghvi17** [comment]
So it turns out that Github doesn't want to let me push updates to workflows from my machine, for some reason.  Those changes are in #158 - if that's merged, I'll rebase this on master.

**Comment by asinghvi17** [comment]
Superseded by #158

**Pull request closed without merging**